### ISSUE TITLE
Potential Issue with Buy orders slippage and fee (for fee=0)

### ISSUE THREAD

**Issue by anxolin** [cowprotocol/cowswap#3503]
**Description**
I believe after the introduction for fee=0 we might have an issue with the slippage calculation.

TLDR:  We apply the slippage calculation to the amount after fee, this makes the slippage bigger than needed. 

**How to Reproduce**
1. (https://dev.swap.cow.fi/#/1/swap/WETH/DAI
2. Set your slippage tolerance to 50% (for dramatic effect)
3. Check the values you get buying 15 DAI in mainnet using USDC
4. You will see the MAX SENT (inc fee) value is bigger than what you might think

**Expected behavior**
Not clear!

We can decide to apply the slippage before or after the fee has been applied. 
* **Before**:  The fee is in theory independent. So I 

**Screenshots**

**Additional context**
Also, if it's relevant for the described byg, add information like the browser or OS.

**Comment by alfetopito** [comment]
Small typo. Should be `option 2` instead of `1`

![image](https://github.com/cowprotocol/cowswap/assets/43217/682539f6-bfbe-4087-ac0c-019d1d27bf39)
